### PR TITLE
.github: Clean up cilium-cli action usages

### PR DESCRIPTION
**Pull request by michi-covalent**
- Install cilium-cli in the default location to be consistent with other workflows. When cilium/design-cfps#9 eventually gets implemented, we might want to put the cilium-cli code under cilium-cli/ directory.
- For conformance-ginkgo.yaml, use the cilium-cli action to install cilium-cli in /host directory.

### CODE REVIEW

File: .github/workflows/conformance-ginkgo.yaml [hunk 1]
```diff
@@ -300,7 +300,7 @@ jobs:
           cmd: |
             git config --global --add safe.directory /host
             mv /host/helm /usr/bin
-            mv /host/cilium-cli /usr/bin
+            mv /host/cilium /usr/bin/cilium-cli
```
Comment (michi-covalent) [inline .github/workflows/conformance-ginkgo.yaml:303]: looking at recent failures with scheduled runs (like [this one](https://github.com/cilium/cilium/actions/runs/9782530035)), it doesn't look like sysdump is being collected. @aanm did we accidentally stop running `cilium sysdump` command with this pull request? https://github.com/cilium/cilium/pull/26143

**Review by julianwiedmann — APPROVED** [review]

File: .github/workflows/conformance-ginkgo.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/conformance-ginkgo.yaml:303]: @michi-covalent I'm not sure if I follow. We do have sysdumps on some test failures, no?

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Review by nebril — APPROVED** [review]

File: .github/workflows/conformance-ginkgo.yaml (hunk 1 above)
Comment (michi-covalent) [reply .github/workflows/conformance-ginkgo.yaml:303]: i only checked like 2 archives from failed runs so maybe, but i couldn't find any step that runs the sysdump command in https://github.com/cilium/cilium/blob/main/.github/workflows/conformance-ginkgo.yaml 💭

**Pull request merged**